### PR TITLE
microoptimizations for VERIFY() and friends

### PR DESCRIPTION
**Pull request by adamdmoss**
- add branch hints
- constify the intermediate evaluations of left/right params in VERIFY3*()

Minor stuff, but why not - the branch hints especially can help keep the spew-code off of hot paths in case the compiler is otherwise confused.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by freqlabs** [comment]
Mind taking a look at these on the FreeBSD side as well?

**Comment by adamdmoss** [comment]
> Mind taking a look at these on the FreeBSD side as well?

Sure!

**Comment by freqlabs** [comment]
Checkstyle why 🤦

**Review by freqlabs — APPROVED** [review]
The code looks good!

File: include/os/freebsd/spl/sys/debug.h [hunk 1]
```diff
@@ -74,9 +74,9 @@ void spl_dumpstack(void);
 	    "%s", "VERIFY(" #cond ") failed\n"))
 
 #define	VERIFY3B(LEFT, OP, RIGHT)	do {				\
-		boolean_t _verify3_left = (boolean_t)(LEFT);		\
-		boolean_t _verify3_right = (boolean_t)(RIGHT);		\
-		if (!(_verify3_left OP _verify3_right))			\
+		const boolean_t _verify3_left = (boolean_t)(LEFT);		\
+		const boolean_t _verify3_right = (boolean_t)(RIGHT);		\
+		if (unlikely(!(_verify3_left OP _verify3_right)))			\
```
Comment (behlendorf) [inline include/os/freebsd/spl/sys/debug.h:79]: nit: it's hard to tell in the GitHub review interface, but let's make sure the trailing `\`'s stay aligned.

**Review by behlendorf — APPROVED** [review]
Good thought!  Looks good.

File: include/os/freebsd/spl/sys/debug.h (hunk 1 above)
Comment (adamdmoss) [reply include/os/freebsd/spl/sys/debug.h]: I'm pretty sure the trailing \'s weren't aligned to begin with - at least in my editor with tabwidth=4 - but they should be now!

**Pull request merged**